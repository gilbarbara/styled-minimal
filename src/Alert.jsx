import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
