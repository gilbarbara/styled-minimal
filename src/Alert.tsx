import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
