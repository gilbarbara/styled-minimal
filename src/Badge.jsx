import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
