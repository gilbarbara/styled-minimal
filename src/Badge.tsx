import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
