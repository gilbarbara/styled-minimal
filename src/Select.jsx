import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { sizeType } from './utils/propTypes';
import { FormStyles } from './utils/system';

export const StyledSelect = styled.select`
  ${FormStyles.select};
  ${FormStyles.pseudo};
`;

const Select = ({ children, ...props }) => (
  <StyledSelect {...props}>{children}</StyledSelect>
);

Select.propTypes = {
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  multiple: PropTypes.bool,
  required: PropTypes.bool,
  size: PropTypes.number,
  sizing: sizeType,
};

Select.defaultProps = {
  sizing: 'md',
};

export default Select;
