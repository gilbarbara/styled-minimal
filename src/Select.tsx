import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, pseudoStyles, sx, system } from './utils/system';

import { SelectProps } from './types';

export const StyledSelect = styled('select')<SelectProps>(
  props => {
    const { multiple, sizing = 'md', valid } = props;
    const {
      backgroundColor,
      color,
      borderColor,
      borderRadius,
      borderWidth,
      fontSize,
      height,
      lineHeight,
      padding,
      validation,
    } = getTheme(props, 'select');

    const currentBgColor = backgroundColor;
    let currentBorderColor = borderColor;

    if (valid) {
      currentBorderColor = validation.valid;
    } else if (valid === false) {
      currentBorderColor = validation.invalid;
    }

    const multipleCheckedColor = currentBorderColor;

    return css`
      ${baseStyles};
      background-color: ${currentBgColor};
      border: ${px(borderWidth)} solid ${currentBorderColor};
      border-radius: ${px(borderRadius)};
      color: ${color};
      display: block;
      ${!multiple ? `height: ${px(height[sizing])}` : ''};
      font-family: inherit;
      font-size: ${px(fontSize[sizing])};
      line-height: ${lineHeight};
      ${!multiple ? `padding: ${px(padding[sizing])};` : ''};
      white-space: nowrap;
      width: 100%;

      > option {
        background-color: ${currentBgColor};
        color: ${color};
        font-size: ${px(fontSize[sizing])};
        line-height: ${lineHeight};
        padding: ${px(padding[sizing])};
      }

      &[multiple] {
        background-color: ${currentBgColor};

        option:checked {
          background: ${`${multipleCheckedColor} linear-gradient(0deg, ${multipleCheckedColor} 0%, ${multipleCheckedColor} 100%)`};
          font-weight: bold;
        }
      }
    `;
  },
  pseudoStyles,
  sx,
  system,
);

export const Select = React.forwardRef<HTMLSelectElement, SelectProps>((props, ref) => (
  <StyledSelect ref={ref} {...props} />
));

Select.defaultProps = {
  sizing: 'md',
};
