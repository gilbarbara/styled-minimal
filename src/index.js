export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
