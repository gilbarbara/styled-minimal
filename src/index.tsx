export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
