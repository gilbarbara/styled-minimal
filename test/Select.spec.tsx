import * as React from 'react';
import { render } from '@testing-library/react';

import { Select } from '../src/Select';

const Fixture = props => (
  <Select {...props}>
    <option value="male">Male</option>
    <option value="female">Female</option>
    <option value="other">Other</option>
  </Select>
);

describe('Select', () => {
  it('should render properly', () => {
    const { container } = render(<Fixture />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly with validation', () => {
    const { container } = render(
      <div>
        <Fixture valid />
        <Fixture valid={false} />
      </div>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
