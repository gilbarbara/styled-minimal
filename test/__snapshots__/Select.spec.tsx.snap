// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 2px;
  color: #2E2E2E;
  display: block;
  height: 34px;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
  white-space: nowrap;
  width: 100%;
}

.c0 > option {
  background-color: #fff;
  color: #2E2E2E;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
}

.c0[multiple] {
  background-color: #fff;
}

.c0[multiple] option:checked {
  background: #ccc linear-gradient(0deg,#ccc 0%,#ccc 100%);
  font-weight: bold;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c0:read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:disabled {
  background-color: #ececec;
  color: #7a7a7a;
  border-color: #ececec;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:focus {
  outline-color: #fff;
  outline-offset: 1px;
}

.c0:required:not(:valid) {
  border-color: #999;
}

.c0:required:not(:valid)::-webkit-input-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid):-moz-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid)::-moz-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid):-ms-input-placeholder {
  color: #b3b3b3;
}

<select
  class="c0"
>
  <option
    value="male"
  >
    Male
  </option>
  <option
    value="female"
  >
    Female
  </option>
  <option
    value="other"
  >
    Other
  </option>
</select>
`;

exports[`Select should render properly with validation 1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #fff;
  border: 1px solid #3ac200;
  border-radius: 2px;
  color: #2E2E2E;
  display: block;
  height: 34px;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
  white-space: nowrap;
  width: 100%;
}

.c0 > option {
  background-color: #fff;
  color: #2E2E2E;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
}

.c0[multiple] {
  background-color: #fff;
}

.c0[multiple] option:checked {
  background: #3ac200 linear-gradient(0deg,#3ac200 0%,#3ac200 100%);
  font-weight: bold;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c0:read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:disabled {
  background-color: #ececec;
  color: #7a7a7a;
  border-color: #ececec;
}

.c0::-webkit-input-placeholder {
  color: #aeaeae;
}

.c0:-moz-placeholder {
  color: #aeaeae;
}

.c0::-moz-placeholder {
  color: #aeaeae;
}

.c0:-ms-input-placeholder {
  color: #aeaeae;
}

.c0:focus {
  outline-color: #fff;
  outline-offset: 1px;
}

.c0:required:not(:valid) {
  border-color: #999;
}

.c0:required:not(:valid)::-webkit-input-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid):-moz-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid)::-moz-placeholder {
  color: #b3b3b3;
}

.c0:required:not(:valid):-ms-input-placeholder {
  color: #b3b3b3;
}

.c1 {
  box-sizing: border-box;
  background-color: #fff;
  border: 1px solid #f00;
  border-radius: 2px;
  color: #2E2E2E;
  display: block;
  height: 34px;
  font-family: inherit;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
  white-space: nowrap;
  width: 100%;
}

.c1 > option {
  background-color: #fff;
  color: #2E2E2E;
  font-size: 16px;
  line-height: 1;
  padding: 8px;
}

.c1[multiple] {
  background-color: #fff;
}

.c1[multiple] option:checked {
  background: #f00 linear-gradient(0deg,#f00 0%,#f00 100%);
  font-weight: bold;
}

.c1::-webkit-input-placeholder {
  color: #aeaeae;
}

.c1:-moz-placeholder {
  color: #aeaeae;
}

.c1::-moz-placeholder {
  color: #aeaeae;
}

.c1:-ms-input-placeholder {
  color: #aeaeae;
}

.c1:-moz-read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c1:read-only {
  background-color: #fafafa;
  color: #7a7a7a;
}

.c1::-webkit-input-placeholder {
  color: #aeaeae;
}

.c1:-moz-placeholder {
  color: #aeaeae;
}

.c1::-moz-placeholder {
  color: #aeaeae;
}

.c1:-ms-input-placeholder {
  color: #aeaeae;
}

.c1:disabled {
  background-color: #ececec;
  color: #7a7a7a;
  border-color: #ececec;
}

.c1::-webkit-input-placeholder {
  color: #aeaeae;
}

.c1:-moz-placeholder {
  color: #aeaeae;
}

.c1::-moz-placeholder {
  color: #aeaeae;
}

.c1:-ms-input-placeholder {
  color: #aeaeae;
}

.c1:focus {
  outline-color: #fff;
  outline-offset: 1px;
}

.c1:required:not(:valid) {
  border-color: #999;
}

.c1:required:not(:valid)::-webkit-input-placeholder {
  color: #b3b3b3;
}

.c1:required:not(:valid):-moz-placeholder {
  color: #b3b3b3;
}

.c1:required:not(:valid)::-moz-placeholder {
  color: #b3b3b3;
}

.c1:required:not(:valid):-ms-input-placeholder {
  color: #b3b3b3;
}

<div>
  <select
    class="c0"
  >
    <option
      value="male"
    >
      Male
    </option>
    <option
      value="female"
    >
      Female
    </option>
    <option
      value="other"
    >
      Other
    </option>
  </select>
  <select
    class="c1"
  >
    <option
      value="male"
    >
      Male
    </option>
    <option
      value="female"
    >
      Female
    </option>
    <option
      value="other"
    >
      Other
    </option>
  </select>
</div>
`;
